site/content/docs/0.1/components/buttons.md

```diff
@@ -8,4 +8,690 @@ aliases:
+
+{{< example >}}
+<a href="#" class="btn btn-default" tabindex="-1" role="button" aria-disabled="true">Default link</a>
+<a href="#" class="btn btn-strong" tabindex="-1" role="button" aria-disabled="true">Strong link</a>
+{{< /example >}}
+
+<!-- OUDS mod: Loading state added -->
+
+## Loading state
+
+The loading state of a button indicates that an action is currently processing or taking place. This state provides feedback to users, enhancing user experience.
+
+To make an animated loading state button, you will add in your standard button:
+- An animated `<svg>`: the animation will differ whether the loading time is known or not.
+- A status message `<span>`: the message must be updated regularly in order to be vocalized by screen readers; if the loading time is defined, you should indicate the progression.
+
+For example, before loading, your button could be like this:
+
+{{< example >}}
+<button type="button" class="btn btn-default" id="loading-btn-0">
+  Download file
+  <svg viewbox='0 0 40 40' xmlns='http://www.w3.org/2000/svg' class="loader d-none" aria-hidden="true">
+    <circle class="loader-inner" cx="20" cy="20" r="17"></circle>
+  </svg>
+  <span role="status" id="loading-btn-msg-0" class="visually-hidden d-none"></span>
+</button>
+{{< /example >}}
+
+When the loading starts, you will have to:
+- Add the class `.loading-indeterminate` (for an unknown loading time) or `.loading-determinate` (for a known loading time) to the button to provide the appropriate styles and animation.
+- Set the disabled state on the button to avoid any unwanted interactions.
+- Remove the `.d-none` from the animated `<svg>` and the status message `<span>`.
+- Update regularly the status message `<span>` containing the loading message.
+
+At the end of the loading, you should:
+
+- Remove the `disabled` attribute to the button to restore its interactivity.
+- Remove the class `.loading-indeterminate` or `.loading-determinate` to restore the button's look.
+- Set a final status message in `<span>`, indicating the loading has ended.
+- Add the `.d-none` class on the animated `<svg>` (we don't add again `.d-none` on the status message `<span>` to let the users know the download has already be done).
+- Put the focus back on the button to ensure similar behavior across browsers and not lose the user after changes.
+
+You can see it live on our [loading buttons live example]({{< docsref "/examples/loading-buttons" >}}).
+
+{{< example class="p-none" >}}
```

Comment:
Maybe, for a better presentation, I would hav split the example in two: buttons and the icon buttons, with an introduction sentence like: "loading state can also be combined with icon buttons". Maybe it is not necessary to have all sorts of icon button, since the logic is always the same... Plus the is also buttons with icon AND text...